x-pack/plugins/data_enhanced/server/collectors/register.ts

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { PluginInitializerContext } from 'kibana/server';
+import { UsageCollectionSetup } from 'src/plugins/usage_collection/server';
+import { fetchProvider } from './fetch';
+
+export interface ReportedUsage {
+  transientCount: number;
+  persistedCount: number;
+  totalCount: number;
+}
+
+export async function registerUsageCollector(
+  usageCollection: UsageCollectionSetup,
+  context: PluginInitializerContext
+) {
+  try {
+    const collector = usageCollection.makeUsageCollector<ReportedUsage>({
+      type: 'search-session',
+      isReady: () => true,
+      fetch: fetchProvider(context.config.legacy.globalConfig$),
+      schema: {
+        transientCount: { type: 'long' },
+        persistedCount: { type: 'long' },
+        totalCount: { type: 'long' },
+      },
+    });
+    usageCollection.registerCollector(collector);
+  } catch (err) {
```

Comment:
NIT: I dont see any reason to have this try/catch in place. `usageCollection` is not optional in this function parameters.